src/Http/RequestFactory.php

```diff
@@ -82,6 +82,7 @@ public function createHttpRequest()
 
 		// path & query
 		$requestUrl = isset($_SERVER['REQUEST_URI']) ? $_SERVER['REQUEST_URI'] : '/';
+		$requestUrl = preg_replace('#^\w++://[^/]++#', '', $requestUrl);
```

Comment:
See [how Symfony handles this](https://github.com/symfony/http-foundation/blob/fac93f4c3c74c3e4609f98c2b98918dc6f1c4c3c/Request.php#L1719-L1725). They replace it only if it matches current schema and host. But I haven't seen any specification which would define how this should actually be handled.